coderd/database/custom_queries.go

```diff
@@ -0,0 +1,62 @@
+package database
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/lib/pq"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/coderd/rbac"
+)
+
+type customQuerier interface {
+	AuthorizedGetWorkspaces(ctx context.Context, arg GetWorkspacesParams, authorizedFilter rbac.AuthorizeFilter) ([]Workspace, error)
+}
+
+// AuthorizedGetWorkspaces returns all workspaces that the user is authorized to access.
+// This code is copied from `GetWorkspaces` and adds the authorized filter WHERE
+// clause.
+func (q *sqlQuerier) AuthorizedGetWorkspaces(ctx context.Context, arg GetWorkspacesParams, authorizedFilter rbac.AuthorizeFilter) ([]Workspace, error) {
```

Comment:
I might change the wording of this. Prefixing with `Authorized` makes me think that I'm authorized to get workspaces, not that the workspaces returned I'm authorized to access.

Thoughts on `GetAuthorizedWorkspaces` instead?

Reply:
Is it possible for us to get rid of `GetWorkspaces` in its entirety? There doesn't seem much of a point in getting workspaces that the user doesn't have access to.